tests/components/switchbot/test_cover.py

```diff
@@ -0,0 +1,292 @@
+"""Test the switchbot covers."""
+
+from unittest.mock import AsyncMock, Mock, patch
+
+import pytest
+
+from homeassistant.components.cover import (
+    ATTR_CURRENT_POSITION,
+    ATTR_CURRENT_TILT_POSITION,
+    ATTR_POSITION,
+    ATTR_TILT_POSITION,
+    DOMAIN as COVER_DOMAIN,
+    CoverState,
+)
+from homeassistant.components.switchbot.const import DOMAIN
+from homeassistant.const import (
+    ATTR_ENTITY_ID,
+    CONF_ADDRESS,
+    CONF_NAME,
+    CONF_SENSOR_TYPE,
+    SERVICE_CLOSE_COVER,
+    SERVICE_CLOSE_COVER_TILT,
+    SERVICE_OPEN_COVER,
+    SERVICE_OPEN_COVER_TILT,
+    SERVICE_SET_COVER_POSITION,
+    SERVICE_SET_COVER_TILT_POSITION,
+    SERVICE_STOP_COVER,
+    SERVICE_STOP_COVER_TILT,
+)
+from homeassistant.core import HomeAssistant, State
+from homeassistant.setup import async_setup_component
+
+from . import WOBLINDTILT_SERVICE_INFO, WOCURTAIN3_SERVICE_INFO
+
+from tests.common import MockConfigEntry
+from tests.components.bluetooth import inject_bluetooth_service_info
+
+
+@pytest.mark.usefixtures("entity_registry_enabled_by_default")
+async def test_curtain3_setup(hass: HomeAssistant) -> None:
+    """Test setting up the Curtain3."""
+    await async_setup_component(hass, DOMAIN, {})
+    inject_bluetooth_service_info(hass, WOCURTAIN3_SERVICE_INFO)
+
+    entry = MockConfigEntry(
+        domain=DOMAIN,
+        data={
+            CONF_ADDRESS: "aa:bb:cc:dd:ee:ff",
+            CONF_NAME: "test-name",
+            CONF_SENSOR_TYPE: "curtain",
+        },
+        unique_id="aabbccddeeff",
+    )
+    entry.add_to_hass(hass)
+
+    mock_data = State("cover.test_name", "open", {ATTR_CURRENT_POSITION: 50})
+    with patch(
+        "homeassistant.helpers.restore_state.RestoreEntity.async_get_last_state",
+        new=AsyncMock(return_value=mock_data),
+    ) as mock_async_get_last_state:
```

Comment:
We shouldn't patch HA internals in tests.

Please use the `mock_restore_cache` fixture from `tests.common` instead